mne/viz/backends/_qt.py

```diff
@@ -280,8 +317,158 @@ def _status_bar_add_progress_bar(self, stretch=0):
         return widget
 
 
+class _QtPlayback(_AbstractPlayback):
+    def _playback_initialize(self, func, timeout):
+        self.figure.plotter.add_callback(func, timeout)
+
+
+class _QtMplCanvas(_AbstractMplCanvas):
+    def __init__(self, brain, width, height, dpi):
+        from matplotlib import rc_context
+        from matplotlib.figure import Figure
+        if brain.separate_canvas:
+            parent = None
+        else:
+            parent = brain._renderer._window
+        # prefer constrained layout here but live with tight_layout otherwise
+        context = nullcontext
+        extra_events = ('resize',)
+        try:
+            context = rc_context({'figure.constrained_layout.use': True})
+            extra_events = ()
+        except KeyError:
+            pass
+        with context:
+            self.fig = Figure(figsize=(width, height), dpi=dpi)
+        from PyQt5 import QtWidgets
+        from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+        self.canvas = FigureCanvasQTAgg(self.fig)
+        self.canvas.setParent(parent)
+        FigureCanvasQTAgg.setSizePolicy(
+            self.canvas,
+            QtWidgets.QSizePolicy.Expanding,
+            QtWidgets.QSizePolicy.Expanding
+        )
+        FigureCanvasQTAgg.updateGeometry(self.canvas)
+        self.manager = None
+        self.axes = self.fig.add_subplot(111)
+        self.axes.set(xlabel='Time (sec)', ylabel='Activation (AU)')
+        self.brain = brain
+        self.time_func = brain.callbacks["time"]
+        for event in ('button_press', 'motion_notify') + extra_events:
+            self.canvas.mpl_connect(
+                event + '_event', getattr(self, 'on_' + event))
```

Comment:
Not DRY. Let's refactor somehow.

For example, you could either have this code in the abstract class `__init__`, and have that `__init__` call the abstract method `self._create_mpl_canvas(...)`, which for this class does the necessary FigureCanvasQtAgg gymnastics, and for the _IpyMplCanvas does the IPython stuff